nncf/openvino/graph/layer_attributes.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Optional
+
+from nncf.common.graph.layer_attributes import BaseLayerAttributes
+from nncf.common.graph.layer_attributes import ConvolutionLayerAttributes
+from nncf.common.graph.layer_attributes import GenericWeightedLayerAttributes
+from nncf.openvino.graph.metatypes.openvino_metatypes import OVConvolutionMetatype
+from nncf.openvino.graph.metatypes.openvino_metatypes import OVDepthwiseConvolutionMetatype
+from nncf.openvino.graph.metatypes.openvino_metatypes import OVGroupConvolutionMetatype
+
+
+class OVConstantLayerAttributesContainer(BaseLayerAttributes):
+    """
+    This class stores mapping weights port indices to constant name and shape.
+    For MatMul layers, it also stores transpose parameter and shape for activations.
+    """
+
+    def __init__(
+        self,
+        const_attrs: Dict[int, Any],
+        common_layer_attrs: Optional[Dict[int, BaseLayerAttributes]] = None,
+        act_attrs: Optional[Dict[Any, Any]] = None,
+    ):
+        """
+        :param const_attrs: Map of weights port ID to corresponding const attributes.
+        :param act_attrs: Activation attributes.
+        """
+        self._const_attrs = const_attrs
+        self.common_layer_attrs = {} if common_layer_attrs is None else common_layer_attrs
+        self._act_attrs = {} if act_attrs is None else act_attrs
+
+    @property
+    def const_attrs(self):
+        return self._const_attrs if self._const_attrs != {} else None
+
+    @property
+    def act_attrs(self):
+        return self._act_attrs if self._act_attrs != {} else None
+
+    def get_const_port_ids(self) -> List[int]:
+        """
+        Returns indices of input ports corresponding to the constant nodes.
+
+        :returns: List of input port indices with constants.
+        """
+        return list(self.const_attrs.keys())
+
+
+def get_weighted_layer_attributes(ov_node, ov_metatype, constant_attributes: Dict[str, Any]):
+    retval = {}
+    for port_id, attrs in constant_attributes.items():
+        if ov_metatype in [OVConvolutionMetatype, OVDepthwiseConvolutionMetatype, OVGroupConvolutionMetatype]:
+            node_attrs = ov_node.get_attributes()
+            kwargs = {
+                "weight_requires_grad": False,
```

Comment:
Can we avoid adding this parameter to post training backends?

Reply:
This parameter is present in common `ConvolutionalLayerAttributes`, I afraid we couldn't get rid of them easily